### PR TITLE
Move files over

### PR DESCRIPTION
**Pull request by nnegrey**
Moving the samples from https://github.com/GoogleCloudPlatform/firebase-appengine-backend to this repo.

### CODE REVIEW

File: flexible/mobile-backend/src/main/webapp/Dockerfiles [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FROM gcr.io/google_appengine/jetty9-compat:githubheadasync
```
Comment (lesv) [inline flexible/mobile-backend/src/main/webapp/Dockerfiles]: That's not really maintained.

File: flexible/mobile-backend/src/main/java/com/google/cloud/solutions/flexenv/common/Message.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 Google LLC
```
Comment (lesv) [inline flexible/mobile-backend/src/main/java/com/google/cloud/solutions/flexenv/common/Message.java]: You mentioned you were moving, so this is probably ok, but it seems wrong to checkin a new file w/ this date.

File: flexible/mobile-backend/pom.xml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>com.google.appengine</groupId>
+      <artifactId>appengine-api-1.0-sdk</artifactId>
+      <version>${appengine.target.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>servlet-api</artifactId>
+      <version>2.5</version>
```
Comment (lesv) [inline flexible/mobile-backend/pom.xml]: that's kind of old

Comment (lesv) [reply flexible/mobile-backend/pom.xml]: 3.1.0 and a different name is expected here.

File: flexible/mobile-backend/src/main/webapp/Dockerfiles (hunk 1 above)
Comment (nnegrey) [reply flexible/mobile-backend/src/main/webapp/Dockerfiles]: I'll check with the author on this.

File: flexible/mobile-backend/src/main/java/com/google/cloud/solutions/flexenv/common/Message.java (hunk 2 above)
Comment (nnegrey) [reply flexible/mobile-backend/src/main/java/com/google/cloud/solutions/flexenv/common/Message.java]: Updated

File: flexible/mobile-backend/pom.xml (hunk 3 above)
Comment (nnegrey) [reply flexible/mobile-backend/pom.xml]: Updated

**Comment by nnegrey** [comment]
These should be good to be moved over now.

File: flexible/mobile-backend/pom.xml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
+    <version>1.0.11</version>
+  </parent>
+
+  <properties>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+
+    <appengine.target.version>1.9.30</appengine.target.version>
```
Comment (lesv) [inline flexible/mobile-backend/pom.xml]: That is really old

Comment (lesv) [reply flexible/mobile-backend/pom.xml]: While this works and will get updated by DPEBot, we moved to putting the versions explicitly in the dependencies as it's easier for the user to figure out.

File: flexible/mobile-backend/pom.xml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+              <filtering>true</filtering>
+              <targetPath>WEB-INF</targetPath>
+            </resource>
+          </webResources>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>1.3.2</version>
```
Comment (lesv) [inline flexible/mobile-backend/pom.xml]: 2.0.0 or so is current.

File: flexible/mobile-backend/pom.xml (hunk 4 above)
Comment (lesv) [reply flexible/mobile-backend/pom.xml]: It probably shouldn't even be used on Flex as compat is deprecated if I recall.

File: flexible/mobile-backend/src/main/webapp/Dockerfiles (hunk 1 above)
Comment (lesv) [reply flexible/mobile-backend/src/main/webapp/Dockerfiles]: Still not maintained.  It probably should be gcr.io/google_appengine/jetty9-compat or gcr.io/jetty9 (nothing in 2 years)or gcr.io/jetty

gcr.io/jetty-compat:release-20190619
or
gcr.io/jetty:2019-06-25-204809 (aka :9)

Comment (lesv) [reply flexible/mobile-backend/src/main/webapp/Dockerfiles]: githubheadasync is very old

Comment (ricalo) [reply flexible/mobile-backend/src/main/webapp/Dockerfiles]: I tested with:
* `gcr.io/google-appengine/jetty`
* `gcr.io/jetty`

The sample works with both, but I rather use `gcr.io/google-appengine/jetty` which is specified in the docs of the [Java/Jetty runtime](https://cloud.google.com/appengine/docs/flexible/java/dev-jetty9).

File: flexible/mobile-backend/pom.xml (hunk 5 above)
Comment (ricalo) [reply flexible/mobile-backend/pom.xml]: Yes, `2.0.0` is current. However, you need some extra configuration to `pom.xml` to move to that version. Check the additional `deploy.projectId` and `deploy.version` elements in the plugin configuration:

```xml
<plugin>
  <groupId>com.google.cloud.tools</groupId>
  <artifactId>appengine-maven-plugin</artifactId>
  <version>2.0.0</version>
  <configuration>
    <promote>true</promote>
    <deploy.projectId>playchat-241319</deploy.projectId>
    <deploy.version>sample</deploy.version>  </configuration>
</plugin>
```

File: flexible/mobile-backend/src/main/webapp/Dockerfiles (hunk 1 above)
Comment (nnegrey) [reply flexible/mobile-backend/src/main/webapp/Dockerfiles]: @ricalo, should this change from 
`FROM gcr.io/google_appengine/jetty9-compat:githubheadasync`
To
`FROM gcr.io/google_appengine/jetty9`?

Comment (ricalo) [reply flexible/mobile-backend/src/main/webapp/Dockerfiles]: Almost, it's `gcr.io/google-appengine/jetty` (without the trailing 9).

Comment (nnegrey) [reply flexible/mobile-backend/src/main/webapp/Dockerfiles]: Updated

File: flexible/mobile-backend/pom.xml (hunk 5 above)
Comment (nnegrey) [reply flexible/mobile-backend/pom.xml]: @ricalo, do you think it should be updated and have the user set the `projectId` field?

Comment (ricalo) [reply flexible/mobile-backend/pom.xml]: Mmm... on second thought, the `projectId` could be set using:

```shell
$ gcloud config set project [project-id]
```
which is already specified in the original README. The `deploy.version` element is the piece that was missing from the original instructions.

It looks to me like you can hardcode the version in this sample. Using the same version means that the app is overwritten if users redeploy the sample, which should be fine in this scenario.

**Comment by lesv** [comment]
Status update please?  can we close this?

**Comment by nnegrey** [comment]
I think we should probably just close this as these use a lot of old stuff. @ricalo thoughts?

**Comment by ricalo** [comment]
Closing sounds appropriate.

**Pull request closed without merging**